rtos/TARGET_CORTEX/TARGET_CORTEX_M/mbed_rtx_fault_handler.c

```diff
@@ -15,104 +15,106 @@
  */
 
 #include "rtx_os.h"
-#include "mbed_rtx.h"
-#include "mbed_rtx_fault_handler.h"
-#include "hal/serial_api.h"
+#include "device.h"
+#include "platform/mbed_error.h"
+#include "platform/mbed_error_report.h"
 
 #ifndef MBED_FAULT_HANDLER_DISABLED
-//Global for populating the context in exception handler
-mbed_fault_context_t mbed_fault_context;
+#include "mbed_rtx_fault_handler.h"
 
-//Structure to capture the context
-void fault_print_init(void);
-void fault_print_str(char *fmtstr, uint32_t *values);
-void hex_to_str(uint32_t value, char *hex_star);
+//Functions Prototypes
```

Comment:
All this code restructuring and changes in fault handler should be a separate PR.